File: command/kotlin/SellStock.kt
package Command

// Nesnelere dönüştürdüğümüz her bir isteğe denk gelmektedir.
// Command (IOrderCommand) arayüzünü uygular.
// UML diyagramındaki ConcreteCommand sınıfına denk gelmektedir.
internal class SellStock(private val _stockManager: StockManager) : IOrderCommand {

    override fun execute() {
        _stockManager.sell()
    }
}

File: command/python/sell_stock.py
from order_command import OrderCommand
from stock_manager import StockManager


class SellStock(OrderCommand):
    """
    Nesnelere dönüştürdüğümüz her bir isteğe denk gelmektedir.
    Command (IOrderCommand) arayüzünü uygular.
    UML diyagramındaki ConcreteCommand sınıfına denk gelmektedir.
    """

    def __init__(self, stock_manager: StockManager):
        self._stock_manager = stock_manager

    def execute(self):
        self._stock_manager.sell()
